homeassistant/components/fitbit/sensor.py

```diff
@@ -432,48 +363,58 @@ def __init__(
                     measurement_system = FITBIT_MEASUREMENTS["en_US"]
             unit_type = measurement_system[split_resource[-1]]
         self._unit_of_measurement = unit_type
-        self._state = 0
+        self._state: str | None = None
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the sensor."""
         return self._name
 
     @property
-    def state(self):
+    def state(self) -> str | None:
         """Return the state of the sensor."""
         return self._state
 
     @property
-    def unit_of_measurement(self):
+    def unit_of_measurement(self) -> str | None:
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
     @property
-    def icon(self):
+    def icon(self) -> str | None:
         """Icon to use in the frontend, if any."""
-        if self.resource_type == "devices/battery" and self.extra:
-            battery_level = BATTERY_LEVELS[self.extra.get("battery")]
-            return icon_for_battery_level(battery_level=battery_level, charging=None)
-        return f"mdi:{FITBIT_RESOURCES_LIST[self.resource_type][2]}"
+        if self.resource_type == "devices/battery" and self.extra is not None:
+            extra_battery = self.extra.get("battery")
+            if extra_battery is not None:
+                battery_level = BATTERY_LEVELS.get(extra_battery)
+                if battery_level is not None:
+                    return icon_for_battery_level(battery_level=battery_level)
+        fitbit_ressource = FITBIT_RESOURCES_LIST.get(self.resource_type)
+        if fitbit_ressource is not None:
+            return f"mdi:{fitbit_ressource[2]}"
+        return None
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, str]:
         """Return the state attributes."""
-        attrs = {}
+        attrs: dict[str, str] = {}
 
         attrs[ATTR_ATTRIBUTION] = ATTRIBUTION
 
-        if self.extra:
-            attrs["model"] = self.extra.get("deviceVersion")
-            attrs["type"] = self.extra.get("type").lower()
+        if self.extra is not None:
+            extra_device_version = self.extra.get("deviceVersion")
+            if extra_device_version is not None:
+                attrs["model"] = extra_device_version
```

Comment:
This looks like a breaking change. I think both attrs should be always set either or `str` or `None`.